pkg/rgrpc/todo/pool/selector.go

```diff
@@ -93,33 +101,39 @@ func (s *Selector[T]) Next(opts ...Option) (T, error) {
 		opt(&options)
 	}
 
-	address := s.id
-	if options.registry != nil {
-		services, err := options.registry.GetService(s.id)
-		if err != nil {
-			return *new(T), fmt.Errorf("%s: %w", s.id, err)
+	target := s.id
+	prefix := strings.SplitN(s.id, ":", 2)[0]
+	switch prefix {
+	case "dns", "unix", "kubernetes":
+		// use target as is
+	default:
+		// use service registry to look up address
+		if options.registry != nil {
+			services, err := options.registry.GetService(s.id)
+			if err != nil {
+				return *new(T), fmt.Errorf("%s: %w", s.id, err)
+			}
+
+			nodeAddress, err := registry.GetNodeAddress(services)
+			if err != nil {
+				return *new(T), fmt.Errorf("%s: %w", s.id, err)
+			}
+			target = nodeAddress
```

Comment:
```suggestion
	case "dns", "unix", "kubernetes":
		       // use target as is
        case options.registry != nil:
                      fallthrough
	default:
                        // use service registry to look up address
			services, err := options.registry.GetService(s.id)
			if err != nil {
				return *new(T), fmt.Errorf("%s: %w", s.id, err)
			}

			nodeAddress, err := registry.GetNodeAddress(services)
			if err != nil {
				return *new(T), fmt.Errorf("%s: %w", s.id, err)
			}
			target = nodeAddress
```
Use fallthrough and specific condition instead of nested if clause.